### ISSUE TITLE
File opts.messagesDir not created because descriptors.length == 0

### ISSUE THREAD

**Issue by googol7** [yahoo/babel-plugin-react-intl#92]
The directory defined in opts.messagesDir is not being created with version > 2.1.4

commit [48b7d47](https://github.com/yahoo/babel-plugin-react-intl/commit/48b7d47f12ff757264bbde3d349a54a7f42d0abb) is the first one that doesn’t work anymore.

created a repo here for you to test:
https://github.com/googol7/test-babel-plugin-react-intl

babel-plugin-react-intl@2.1.4 works
babel-plugin-react-intl@2.2.0 doesn’t work anymore
babel-plugin-react-intl@2.3.0 doesn’t work either

please try these commands:
```
npm run translate
```
or
```
./translate.sh
```

I suppose it’s connected to these issues where the command is run twice and the second time `descriptors.length == 0`:
https://github.com/yahoo/babel-plugin-react-intl/pull/28
https://github.com/yahoo/babel-plugin-react-intl/issues/25

Is there something wrong in my configuration or is there a bug with versions > 2.1.4?

**Comment by ericf** [comment]
There hasn't been any changes to the code that actually writes out of the `.json` files.

**Comment by googol7** [comment]
The test [repo](https://github.com/googol7/test-babel-plugin-react-intl) I made is very reduced and set up according to the setup instructions.

Please try this to reproduce the problem:

```
git clone git@github.com:googol7/test-babel-plugin-react-intl.git
cd test-babel-plugin-react-intl/
yarn install
npm run translate
```

an error occurs because the directory extracted-messages was not created.

then downgrade:
```
yarn add babel-plugin-react-intl@2.1.4
npm run translate
```
voilà - it works - the directory extracted-messages was created.

after upgrading with
```
yarn add babel-plugin-react-intl@2.3.0
npm run translate
```
the directory will not be created.

**Comment by googol7** [comment]
@ericf could you please have a quick look at the repo https://github.com/googol7/test-babel-plugin-react-intl what’s wrong with my setup?

**Comment by chemoish** [comment]
I am having a similar issue with `babel#6.9.1` and `babel-plugin-react-intl#2.3.0` where no messages are being generated.

I will roll back and see when this issue started.

**Comment by chemoish** [comment]
@ericf I can confirm that as of `2.2.0`, messages are no longer generated. Messages now are `[]`—like @googol7 says, they work fine in `2.1.4`.

**Comment by ericf** [comment]
I haven't had a chance to look at this yet. If you guys see anything in the commits between the versions that work and don't work let me know, or send a PR if you find a fix.

**Comment by ericf** [comment]
I see the README docs encourage someone to end up adding the plugin twice, once via `.babelrc` and again via the command line `--plugins react-intl`. I should update the docs to remove the command line usage since that's never going to yield what someone wants and cause the plugin to be added multiple times.

If you guys remove `--plugins react-intl` from your babel command line the problem should be resolved. And I'll look to improve things in the plugin to recognize this state and try to do something better.

**Comment by googol7** [comment]
@eric thanks!

**Comment by ericf** [comment]
I was able to reproduce and `--plugins react-intl` CLI option causes the issue because it adds an instance of the babel plugin without any config options set. When the second instance of the plugin runs that's defined and configured in `.babelrc` it becomes a noop. I can produce a warning when multiple instances of the babel plugin is detected, I'll try to see if I can warn about `--plugins` usage since it's most likely not what people want since you can't configure it to do anything useful.

**Comment by ericf** [comment]
Alright I figured out a fix for this issue without warning. It Just Works™

**Comment by ericf** [comment]
@googol7 @chemoish this should be fix in `v2.3.1`, give it a try and let me know!

**Comment by chemoish** [comment]
@ericf not sure what changed, but your comment above makes sense.

I tried to use `--plugins react-intl` and get no warnings or files generated. 

When I just leverage `.babelrc`, using the code I listed above, it works the same between `2.3.0` and `2.3.1`—files are generated.

Either way I am updating my code to match the snippet above since what you said makes sense.

**Comment by ericf** [comment]
`--plugins react-intl` won't yield anything useful.

If you configured the plugin with `.babelrc` and **also** included it via `--plugins` — which causes it to be included twice — it'll now output the messages that were extracted from this first instance of the plugin.

Essentially what I did was allow the plugin to work as expect, even when it's being included more that one time. I was able to do this without having to add warnings. There's also no way for the to detect in the plugin whether it was added via `.babelrc` or `--plugins`.

**Comment by chemoish** [comment]
Right i was using `--plugin react-intl` *WITH* `.babelrc`—but no worries, i will just use `.babelrc` on its own.

Thanks for the help!

**Comment by ericf** [comment]
Not sure why. I tested it this way and saw the issue, then with my changes the issue was fixed.

**Comment by googol7** [comment]
@ericf your fix in `v2.3.1` works. thanks!
